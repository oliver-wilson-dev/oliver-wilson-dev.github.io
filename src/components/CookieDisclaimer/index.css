:root {
    --cookie-disclaimer-text-colour: #333;
    --cookie-disclaimer-text-font-size: 16px;
}