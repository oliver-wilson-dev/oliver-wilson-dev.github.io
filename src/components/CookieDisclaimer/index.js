import React, { useState, useEffect } from 'react';
import cn from 'classnames';
import styles from './index.css';

const CookieDisclaimer = () => {
  const [acknowledged, setAcknowledged] = useState(false);
  const [loaded, setLoaded] = useState(false);
  const [closed, setClosed] = useState(false);

  useEffect(() => {
    setLoaded(true);
  }, []);

  const onTransitionEnd = () => {
    if (acknowledged) setClosed(true);
  };

  const acknowledgeBanner = () => {
    setAcknowledged(true);
  };

  return !closed && (
    <div
      className={cn(styles.container, {
        [styles.loaded]: loaded && !acknowledged,
        [styles.closed]: acknowledged
      })}
      onTransitionEnd={onTransitionEnd}
    >
      <div className={styles.positionWrapper}>
        <div className={cn(styles.cookieEmoji, styles.notificationElement)} role="img" aria-label="cookie emoji" />
        <span className={cn(
          styles.message,
          styles.notificationElement,
        )}
        >
          Attention! This site uses cookies.
        </span>
        <button className={cn(styles.tick, styles.notificationElement)} type="button" onClick={acknowledgeBanner} />
      </div>
    </div>
  );
};

export default CookieDisclaimer;
