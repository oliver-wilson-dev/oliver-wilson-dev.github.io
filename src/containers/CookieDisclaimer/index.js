import { connect } from 'react-redux';
import CookieDisclaimer from '../../components/CookieDisclaimer';
import { getShowLearnMore } from '../../state/selectors';

const mapStateToProps = state => ({
  showLearnMore: getShowLearnMore(state),
});

export default connect(mapStateToProps)(CookieDisclaimer);
