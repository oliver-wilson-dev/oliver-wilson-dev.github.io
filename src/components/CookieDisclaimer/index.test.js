import React from 'react';
import { shallow } from 'enzyme';

import CookieDisclaimer from '.';

jest.mock('../../containers/CookieDisclaimerMessage', () => {
  const CookieDisclaimerMessage = () => null;

  return CookieDisclaimerMessage;
});

jest.mock('../LearnMoreOverlay', () => {
  const LearnMoreOverlay = () => null;

  return LearnMoreOverlay;
});

const defaultProps = {
  showLearnMore: true
};

const render = (renderMethod = shallow, props = {}) => renderMethod(
  <CookieDisclaimer
    {...defaultProps}
    {...props}
  />
);

describe('Cookie Disclaimer component', () => {
  describe('when learnShowMore is true', () => {
    it('should render correctly', () => {
      expect(render()).toMatchSnapshot();
    });
  });

  describe('when learnShowMore is false', () => {
    it('should render correctly', () => {
      expect(render(shallow, { showLearnMore: false })).toMatchSnapshot();
    });
  });
});
