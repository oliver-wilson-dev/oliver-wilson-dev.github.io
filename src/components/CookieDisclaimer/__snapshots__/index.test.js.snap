// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Disclaimer component when first loaded should render correctly 1`] = `
<div
  className="container"
  onTransitionEnd={[Function]}
>
  <div
    className="positionWrapper"
  >
    <div
      aria-label="cookie emoji"
      className="cookieEmoji notificationElement"
      role="img"
    />
    <span
      className="message notificationElement"
    >
      Attention! This site uses cookies.
    </span>
    <button
      className="tick notificationElement"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Cookie Disclaimer component when first loaded should then apply the loaded class 1`] = `
<CookieDisclaimer>
  <div
    className="container loaded"
    onTransitionEnd={[Function]}
  >
    <div
      className="positionWrapper"
    >
      <div
        aria-label="cookie emoji"
        className="cookieEmoji notificationElement"
        role="img"
      />
      <span
        className="message notificationElement"
      >
        Attention! This site uses cookies.
      </span>
      <button
        className="tick notificationElement"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</CookieDisclaimer>
`;

exports[`Cookie Disclaimer component when first loaded when the user accepts the policy after the transition ends should render null 1`] = `""`;

exports[`Cookie Disclaimer component when first loaded when the user accepts the policy should render correctly 1`] = `
<CookieDisclaimer>
  <div
    className="container closed"
    onTransitionEnd={[Function]}
  >
    <div
      className="positionWrapper"
    >
      <div
        aria-label="cookie emoji"
        className="cookieEmoji notificationElement"
        role="img"
      />
      <span
        className="message notificationElement"
      >
        Attention! This site uses cookies.
      </span>
      <button
        className="tick notificationElement"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</CookieDisclaimer>
`;
